### PR TITLE
Issue 149

### LINKED ISSUE #149: Add functionality to UI to allow admin users to create/update/delete Facilities (Entities)
### User Story
As a Admin User of the HealthCare RollCall UI
I'd like to have the ability to create/update/delete Facilities (Entities)
So that I can manage Facilities with both the endpoint calls and UI

### Acceptance Criteria
- [x] Verify a Facility can be created (admin role only)
- [x] Verify a Facility can be updated (admin role only)
- [x] Verify a Facility can be deleted (admin role only)
- [x] Verify a newly created Facility can be updated (admin role only)
- [x] Verify a newly created Facility can be deleted (admin role only)
- [x] Verify a newly created Facility can be associated with a Contact (admin role only)
- [x] Verify a newly created Facility can be associated with more than one Contact (admin role only)
- [x] Verify that a Facility's Contacts can be modified (admin role only)
- [x] Verify that a user cannot add a Facility.

### Notes
See wireframes here - https://www.figma.com/file/tjhV8ueayBrTeuqnzKN0EH/Healthcare-Roll-Call?node-id=54%3A68
This is a lot of functionality for one story.  It may need to be broken out into multiple.

### PR DESCRIPTION
**Pull request by revjtanton**
This will add facility add/edit/delete functionality to the front-end.
resolves #149 

## Related PRs

List related PRs against other branches:

| branch          | PR       |
| --------------- | -------- |
| Bmore-Responsive | [link](https://github.com/CodeForBaltimore/Bmore-Responsive/pull/220) |

## Todos

- [x] Tests
- [x] Documentation

## Deploy Notes

You must be on the `1.3.1` version of bmore-res.

## Impacted Areas in Application

List general components of the application that this PR will affect:

- facility add/edit/delete

### CODE REVIEW

File: src/templates/Dashboard.vue [hunk 1]
```diff
@@ -35,12 +35,14 @@
               </b-input-group>
             </b-form-group>
           </b-col>
+          <div v-if="showAdmin" >
+            <b-button v-on:click="editFacility()">Create Facility</b-button>
```
Comment (c-w-allen) [inline src/templates/Dashboard.vue:39]: This function seems to be adding the facility more than editing it?

File: src/templates/Facility.vue [hunk 2]
```diff
@@ -271,7 +254,8 @@ export default {
         covidForm: [
           "Assisted Living Facility",
           "Mixed Housing",
-          "Senior Housing"
+          "Senior Housing",
+          "Test"
```
Comment (c-w-allen) [inline src/templates/Facility.vue:258]: Accidental or intentional?

File: src/templates/FacilityEdit.vue [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+          contacts: []
+        },
+        showAdmin,
+        contactSelectList: [{ value: null, text: "Contacts Linked" }],
+        selectedContactID: null,
+        stateSelected: "MD",
+        stateOptions: [{ value: "MD", text: "Maryland" }],
+        typeSelected: null,
+        typeOptions: [
+          { value: null, text: "Select facility type" },
```
Comment (c-w-allen) [inline src/templates/FacilityEdit.vue:88]: I think you could clean this array generation up a little, but not a big deal

File: src/templates/Dashboard.vue (hunk 1 above)
Comment (revjtanton) [reply src/templates/Dashboard.vue:39]: yeah it was just a copy/paste of another function from another component. I got lazy and didn't re-name it.

File: src/templates/Facility.vue (hunk 2 above)
Comment (revjtanton) [reply src/templates/Facility.vue:258]: accidental!

File: src/templates/FacilityEdit.vue (hunk 3 above)
Comment (revjtanton) [reply src/templates/FacilityEdit.vue:88]: Got rid of test, but this is a placeholder until we add a type endpoint to the API.

Comment (revjtanton) [reply src/templates/FacilityEdit.vue:88]: https://github.com/CodeForBaltimore/Bmore-Responsive/issues/219

**Pull request merged**